hvac/adapters.py

```diff
@@ -96,17 +58,24 @@ def __init__(self, base_uri='http://localhost:8200', token=None, cert=None, veri
             'proxies': proxies,
         }
 
-    def close(self):
-        """Close the underlying Requests session.
+    @staticmethod
+    def urljoin(*args):
+        """Joins given arguments into a url. Trailing and leading slashes are stripped for each argument.
+
+        :param args: Multiple parts of a URL to be combined into one string.
+        :type args: basestring
+        :return: Full URL combining all provided arguments
+        :rtype: basestring
         """
-        self.session.close()
+
+        return '/'.join(map(lambda x: str(x).strip('/'), args))
 
     def get(self, url, **kwargs):
         """Performs a GET request.
 
         :param url: Partial URL path to send the request to. This will be joined to the end of the instance's base_uri
             attribute.
-        :type url: str
+        :type url: basestring
```

Comment:
`basestring` is in python2 to distinguish from unicode, but python3 has no such thing, it's all `str`.  as we discussed offline, defaulting to `str` seems more sensible, and tweak your IDE (if possible) to not make a fuss about an "incorrect" type (like unicode which is actually fine) being passed into the method

Reply:
Feast your eyes on this poetry: b853ae8